### PR TITLE
CVE-2015-6765 and CVE-2011-2786

### PR DESCRIPTION
**Pull request by moseslagoon**
CVE-2015-6765

### CODE REVIEW

**Comment by hnv1002** [comment]
CVE-2015-6765
- Please explain what's UaF and how does it cause the browser to crash?
- typo "tim" in your answer to "How was this vulnerability discovered?" question.
- your answer to "What subsystems was the mistake in" is wrong. A compiler is not part of an application, can't really be categorized as a subsystem in this case. This vulberability is not related to compiler any way.
- Please elaborate more on mistakes section. What led to this vulnerability? What are some mitigation plans?
- upvotes: 4 (This is a critical vulnerability, however seems like a corner case)

CVE-2011-2786
- CWE 20 doesn't seem to apply here because it has nothing to do with input validation. In my opinion, CWE-451 is more appropriate.
- Please rephrase the description rather than using nvd.nist.gov description.
- Fix and VCC have the same commit?? also duplicate vccs.
- I don't see unit tests for this vulnerability
- How was this vulnerability found?
- The subsytem should be speech, speech_input_bubble is really just a feature in this subsystem.
- interesting commit is the same as VCC??
- I think defense in depth principle can be apply here since they forgot to account for the case where speech input bubble popup can be moved outside of the screen
- What are some mitagation plans for this vulnerability? I don't think user would intentionally drag the speech recognition bubble outside of the screen themselves.
- upvotes: 2

**Comment by moseslagoon** [comment]
@hnv1002, Thanks for the feedback.

File: cves/CVE-2011-2786.yml [hunk 1]
```diff
@@ -22,17 +22,21 @@ announced: 2011-08-02 20:55:01.863000000 -04:00
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Google Chrome  does not ensure that the speech-input bubble is 
+  shown on the product's screen, which might make it easier for remote attackers to 
+  make audio recordings via a crafted web page containing an INPUT element.
+  Web page can start speech recognition whenever user clicks anything on web page. 
+  The problem is that Chromium relies on a popup showing that recognition is started. 
+  That popup can be moved to be outside the screen.
```
Comment (andymeneely) [inline cves/CVE-2011-2786.yml]: Good - I could use a little bit of a longer example here.

File: cves/CVE-2011-2786.yml [hunk 2]
```diff
@@ -53,10 +57,13 @@ fixes_vcc_instructions: |
 fixes:
 - :commit: 90e7e1e5afd58e5042bc7431c072f9d4bb497621
   :note: ''
-vccs: []
+vccs: |
+- commit: 90e7e1e5afd58e5042bc7431c072f9d4bb497621
```
Comment (andymeneely) [inline cves/CVE-2011-2786.yml]: VCC and fix shouldn't be the same.

File: cves/CVE-2011-2786.yml [hunk 3]
```diff
@@ -67,29 +74,26 @@ unit_tested:
...
     for this module.
-
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    From looking at the fix code it seems that the code that tested manually and 
+    they also added a new test for fixing this vulnerability.
```
Comment (andymeneely) [inline cves/CVE-2011-2786.yml]: You mean automatically? How do you know it was tested manually?

File: cves/CVE-2011-2786.yml [hunk 4]
```diff
@@ -100,33 +104,30 @@ discovered:
 subsystem:
   question: |
     What subsystems was the mistake in?
-
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: Based on the blog post and commits
```
Comment (andymeneely) [inline cves/CVE-2011-2786.yml]: Can you give a link to the blog post?

File: cves/CVE-2011-2786.yml [hunk 5]
```diff
@@ -100,33 +104,30 @@ discovered:
...
-
     Write a brief (under 100 words) description of why you think this commit was
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-  - commit: 
-    note: 
-  - commit: 
-    note: 
+  - commit: 90e7e1e5afd58e5042bc7431c072f9d4bb497621
```
Comment (andymeneely) [inline cves/CVE-2011-2786.yml]: Need something between the VCC and fix on the file

File: cves/CVE-2015-6765.yml [hunk 6]
```diff
@@ -22,19 +22,21 @@ announced: 2015-12-05 20:59:01.637000000 -05:00
...
   description later to get more technical.
-
   Try to still be specific in your description, but remove Chromium-specific
   stuff. Remove references to versions, specific filenames, and other jargon
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: an error in storing entries and fetchers allows attacker to overload
+description: |
+  an error in storing entries and fetchers allows attacker to overload
```
Comment (andymeneely) [inline cves/CVE-2015-6765.yml]: I need this revised to be cleaner. Give a scenario of how someone would attack this.

File: cves/CVE-2011-2786.yml [hunk 7]
```diff
@@ -185,11 +185,13 @@ mistakes:
...
     Maintainability? Requirements? Miscommunications?
-
     Look at the CWE entry for this vulnerability and examine the mitigations
     they have written there. Are they doing those? Does the fix look proper?
-
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
```
Comment (andymeneely) [inline cves/CVE-2011-2786.yml]: So is this a requirements issue? Design? Doesn't really sound like a coding problem.

Comment (andymeneely) [reply cves/CVE-2011-2786.yml]: UI testing?

File: cves/CVE-2015-6765.yml [hunk 8]
```diff
@@ -192,15 +189,13 @@ mistakes:
...
     Maintainability? Requirements? Miscommunications?
-
     Look at the CWE entry for this vulnerability and examine the mitigations
     they have written there. Are they doing those? Does the fix look proper?
-
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: This was a simple exploitable sandbox escape as they say it. They worked on the fix.
```
Comment (andymeneely) [inline cves/CVE-2015-6765.yml]: I need more here - how was the mistake made? Not so much WHAT the mistake was, but how was it made and not found? How did it slip by?

File: cves/CVE-2011-2786.yml [hunk 9]
```diff
@@ -53,10 +57,13 @@ fixes_vcc_instructions: |
 fixes:
 - :commit: 90e7e1e5afd58e5042bc7431c072f9d4bb497621
   :note: ''
-vccs: []
+vccs: 
+- commit: 90e7e1e5afd58e5042bc7431c072f9d4bb497621
```
[the remark this answers is not in the document]
Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Thank you for the commands, Professor. Investigated and added them.  I did check the other files as well, so far, these seem to be the only VCCs.

File: cves/CVE-2011-2786.yml (hunk 3 above)
Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Sorry professor, it seems that I misunderstood code for something else to be the unit tests. As Chris in my cohort pointed out; scrutinizing one more time, apparently, no unit tests exist for this vulnerability.

Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: It seems to be regression instead.

File: cves/CVE-2011-2786.yml (hunk 7 above)
Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Now that I think of it, seems to be a mix of requirements and design issue. Fixed the sentence.

**Comment by moseslagoon** [comment]
@hnv1002 thanks for pointing out areas of improvement/fix for **CVE-2011-2786**. I have addressed all of them in the following order. 
- CWE-451 indeed seems to be more appropriate, thus replaced. 
- Reworked on the description
- Fixed the VCCs. I misunderstood this. Thanks to Prof. Meneely for clearing it out. 
- I mistook the regression testing they did after the fix for unit tests. Hence, there seems to be no unit tests for this VCC. Set to false. Noted. 
- Added more elaboration on how the vulnerability was found from the blog posts and commits.
- Speech indeed seems to be the subsystem. 
- I thought this was an interesting commit as it ultimately led to the fix? 
- I think that would be another one of the principle that could be applied too. Hence, added. 
- Added mitigation plans and thought that this was a design problem with a mix of requirements.

File: cves/CVE-2011-2786.yml (hunk 4 above)
Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Elaborated now. Thanks

Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Seems I left the sentence incomplete here. I have updated the actual name of the subsystem

**Comment by moseslagoon** [comment]
@hnv1002 addressed your comments for CVE-2015-6765.yml

File: cves/CVE-2015-6765.yml [hunk 10]
```diff
@@ -60,80 +64,77 @@ fixes:
...
+    They did not consider the implications of the update here. 
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
-
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 5
```
Comment (peckrog) [inline cves/CVE-2015-6765.yml]: Upvote: 2

File: cves/CVE-2011-2786.yml [hunk 11]
```diff
@@ -52,81 +65,81 @@ fixes_vcc_instructions: |
...
+  note: 'They were working on a fix for something else at the time.'
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
-
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 2
```
Comment (peckrog) [inline cves/CVE-2011-2786.yml]: Upvote: 2

Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Updated!

File: cves/CVE-2011-2786.yml [hunk 12]
```diff
@@ -67,66 +74,60 @@ unit_tested:
     Were automated unit tests involved in this vulnerability?
```
[the remark this answers is not in the document]
Comment (moseslagoon) [reply cves/CVE-2011-2786.yml]: Updated:)

**Pull request merged**